Peer.GitHub/GitHubRequestFetcher.cs

```diff
@@ -17,38 +20,50 @@ public class GitHubRequestFetcher : IPullRequestFetcher
         private readonly GraphQLHttpClient _gqlClient;
         private readonly GitHubPeerConfig _config;
         private readonly AsyncLazy<string> _username;
-        private readonly AsyncLazy<GraphQLHttpRequest> _involvesRequest;
-        private readonly AsyncLazy<GraphQLHttpRequest> _reviewRequestedRequest;
         private readonly CancellationTokenSource _cts = new();
 
-        public GitHubRequestFetcher(GraphQLHttpClient client, GitHubPeerConfig gitHubPeerConfig)
+        public GitHubRequestFetcher(
+            GraphQLHttpClient client,
+            GitHubPeerConfig gitHubPeerConfig)
         {
             _gqlClient = client;
             _config = gitHubPeerConfig;
             _username = new AsyncLazy<string>(() => GetUsername(_cts.Token));
-            _involvesRequest = new AsyncLazy<GraphQLHttpRequest>(() => GenerateInvolvesRequest());
-            _reviewRequestedRequest = new AsyncLazy<GraphQLHttpRequest>(() => GenerateReviewRequestedRequest());
         }
 
-        public async Task<IEnumerable<PullRequest>> GetPullRequestsAsync(CancellationToken token = default)
+        public Task<IAsyncEnumerable<PullRequest>> GetPullRequestsAsync(CancellationToken token = default)
         {
             using var registration = token.Register(() => _cts.Cancel());
+            return Task.FromResult(GetPullRequestsImpl(token));
+        }
 
-            var involvesResponse =
-                await _gqlClient.SendQueryAsync<GQL.SearchResult<PRSearch.Result>>(await _involvesRequest, token);
+        private async IAsyncEnumerable<PullRequest> GetPullRequestsImpl([EnumeratorCancellation] CancellationToken token)
+        {
+            if (token.IsCancellationRequested)
+                yield break;
 
-            var reviewRequestedResponse = await _gqlClient.SendQueryAsync<GQL.SearchResult<PRSearch.Result>>(await _reviewRequestedRequest, token);
+            var responses = new List<IAsyncEnumerable<PRSearch.PullRequest>>
+                {
+                    QueryGithubPullRequests(QueryType.Involves, token),
+                    QueryGithubPullRequests(QueryType.TeamRequested, token)
+                }.Merge();
 
-            var deduplicated = involvesResponse.Data.Search.Nodes
-                .Concat(reviewRequestedResponse.Data.Search.Nodes)
-                .DistinctBy(x => x.Id);
+            var deduplicated = responses.Distinct(x => x.Id);
 
-            // todo: Handle errors.
+            var prs = new Dictionary<string, PRSearch.PullRequest>();
+            var prsWithMoreThreads = new List<PRSearch.PullRequest>();
 
-            var prs = deduplicated.ToDictionary(pr => pr.Id);
+            await foreach (var value in deduplicated)
+            {
+                if (value.ReviewThreads.PageInfo.HasNextPage)
+                {
+                    prsWithMoreThreads.Add(value);
```

Comment:
Now that we’re yielding the other ones out I think prs is the same collection as prs with more threads. We can yield out the completed ones in each iteration of the paging loop and ditch the second collection.

Reply:
Actual we can merge the loop behaviours too.

Reply:
Still wanna chat about this but wanna get it in in a different PR?